Fixed memory leak in Frame pointer

yashrajsapra opened the pull request:
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[Issue]

**Description**

Precise description of the changes in this pull request

**Alternative(s) considered**

Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**

Type Choose one: (Bug fix | Feature | Documentation | Testing | Other)

**Screenshots (if applicable)**

**Checklist**

- [x] I have read the [Contribution Guidelines](https://github.com/Apra-Labs/ApraPipes/wiki/Contribution-Guidelines)
- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

yashrajsapra committed:
Fixed memory leak in Frame pointer

mraduldubey commented:
Another issue is with `myOrig`. 
Currently, its working because `FrameFactory::create` always calls the `Frame::Frame(void *buff, size_t size, boost::shared_ptr<FrameFactory> mother):mutable_buffer(buff,size), myOrig(buff), myMother(mother)` that points the myOrig to new memory location given by the `HostAllocator::create` call in the `FrameFactory::create` function. 
This makes sure that even though myOrig is never reset in the `~Frame` - we never get a frame that is pointing to stale memory. But, its a fragile assumption. Also, we use `myOrig` as a way to make changes to the memory allocated for the frame. We create a new Object Pool slot with older myOrig value but reset the FF::myOrig to NULL to make sure destroy() does not clear the memory at the older object pool slot. This needs to be tested in this PR for behavioral changes. 

<img width="1365" height="1013" alt="image" src="https://github.com/user-attachments/assets/df59afb9-ecaf-4ac1-a60c-b1cceb28bbd6" />

mraduldubey commented:
Another alternative for this is overloading the `+=` operator from the `boost:asio:mutable_buffer` (which just increments the data pointer) in the `Frame` class and free the extra chunks there itself. It might be better than the current approach in the time lag between the `+=` and the eventual freeing of the chunks (until destroy() is called).

kumaakh committed:
Delete .clang-format


Filename had \r, made it impossible to checkout on windows

Commit:
improved logging for debugging issues

Commit:
Addresses a common issues observed in multiple projects, Some times we see no logs. It is being caused due to this line where we try to log before Logger::initLogger(..) is called.
The application code should make sure that Logger::initLogger(..) is the first line of application. if it is not, then they should make sure that they are only using std::coud and std::cerr for "Early logs".
This should be improved to be a more permanent solution where Logger::initLogger(..) overwrites previous log instances

yashrajsapra committed:
Add playback speed control for Mp4ReaderSource and related command handling

yashrajsapra committed:
Merge branch 'NVR_Release_Mem_Leak_FIx' of https://github.com/Apra-Labs/ApraPipes into NVR_Release_Mem_Leak_FIx

mraduldubey commented:
Any update on this @yashrajsapra ? why have we not merged this crucial PR ?

yashrajsapra commented:
It's already merged in apranvr codebase @mraduldubey , Will check if main has this change or not
